openwisp_network_topology/visualizer/views.py

```diff
@@ -14,6 +16,12 @@ class TopologyListView(View):
 
     def get(self, request):
         topologies = self.topology_model.objects.filter(published=True)
+        user = request.user
+        if TOPOLOGY_API_AUTH_REQUIRED and not user.is_superuser:
+            auth_perm = check_auth_perm(request)
```

Comment:
I opened this view as an unauthenticated user and I was able to access it anyway

Reply:
You should not be able to do that. do you make sure `TOPOLOGY_API_AUTH_REQUIRED` was `True` before testing?